feat: Add options to specify AWS credential source for RDS functions (#3289)
kanisterio/kanister: docs_new/functions.md

  | ---------- | :------: | ------ | ----------- |
  | instanceID | Yes      | string | ID of RDS instance you want to create snapshot of |
  | dbEngine   | No       | string | Required in case of RDS Aurora instance. Supported DB Engines: `aurora` `aurora-mysql` and `aurora-postgresql` |
  | credentialsSource | No | string | Source for aws credentials. Supported sources: `profile`, `secret`, `serviceaccount`. Default value is `profile` |
  | credentialsSecret | No | string | Secret to get credentials from. Only used with `credentialsSource: secret`. Secret with this name should be referenced in the Actionset |
  | region | No | string | AWS region to use. Derived from profile or serviceaccount if not set |

::: tip NOTE

> `credentialsSource: serviceaccount` uses kanister operator service account. IAM role for service

Reviewer: would it not be a good idea to have another field (`credentialsSA`) like `credentialsSecret`, that can can be used to configured the service account?

Author: Maybe. The idea here was that an AWS IAM admin would allow kanister service account access to RDS by creating a role for it.
Associating a different service account would be better because it's not trivial to assign multiple roles to a single SA, but implementation-wise it's harder because we currently run RDS operation from within kanister controller (in the function code) meaning that we can't change SA on the flight there.

We can also try using AssumeRole and configure which role it will assume. This is going to be more flexible with less code changes.

I'd suggest we keep it for the follow-up PRs.

Reviewer: ok 👍 .

set, `default` DB Subnet group will be used.
:::

::: tip NOTE

> If `credentialsSource` is configured to `profile` (default), the profile used has to be
> an S3 profile in the same region as the database snapshot.
> It's recommended to export to another region. To do that `credentialSource: secret` or `credentialSource: serviceaccount` can be used.

Reviewer: ```suggestion
If `credentialsSource` is configured to `profile` (default behaviour), the profile used has to be
an S3 profile configured with the same region as the database snapshot.
If it's required to export to another region, `credentialSource: secret` or `credentialSource: serviceaccount` can be used.
```

Author: Updated
